pyiceberg/table/snapshots.py

```diff
@@ -226,7 +226,8 @@ def __eq__(self, other: Any) -> bool:
 class Snapshot(IcebergBaseModel):
     snapshot_id: int = Field(alias="snapshot-id")
     parent_snapshot_id: Optional[int] = Field(alias="parent-snapshot-id", default=None)
-    sequence_number: Optional[int] = Field(alias="sequence-number", default=None)
+    # cannot import `INITIAL_SEQUENCE_NUMBER` due to circular import
+    sequence_number: Optional[int] = Field(alias="sequence-number", default=0)
```

Comment:
Is there a reason the default value for the sequence number has to be changed to 0 as opposed to None?

Reply:
According to the spec, https://iceberg.apache.org/spec/#version-2

```
Snapshot JSON:
sequence-number was added and is required; default to 0 when reading v1 metadata
```

Also added this in the PR description

Reply:
Thank you! I missed the part about the V1 spec. Following your suggestion, I added a `field_serializer` for `TableMetadataCommonFields`. This will ensure that the `Snapshot` pydantic object will not have the `sequence-number` field for V1 format